### PR TITLE
upload

### PR DESCRIPTION
**Pull request by MarcinNieradka**
added router requests and Joi schema

### CODE REVIEW

File: models/contacts.js [hunk 1]
```diff
@@ -1,19 +1,123 @@
...
+    console.error('Error occured when trying to read file:'.red, error);
+    throw error;
+  }
+};
 
-const removeContact = async (contactId) => {}
+const writeContactsFile = async contacts => {
+  try {
+    await fs.writeFile(contactsPath, JSON.stringify(contacts, null, 2));
+    // console.log('File has been updated.'.green);
```
Comment (AdamChodowiecGoIt91) [inline models/contacts.js]: unnecessary comment. Please remove all comments from the project

File: models/contacts.js [hunk 2]
```diff
@@ -1,19 +1,123 @@
...
+};
 
-const addContact = async (body) => {}
+const listContacts = async () => {
+  try {
+    const contacts = await readContactsFile();
 
-const updateContact = async (contactId, body) => {}
+    return contacts;
+    // console.log('Contacts list:'.yellow);
```
Comment (AdamChodowiecGoIt91) [inline models/contacts.js]: comments

File: models/contacts.js [hunk 3]
```diff
@@ -1,19 +1,123 @@
...
+};
+
+const getContactById = async contactId => {
+  try {
+    const contacts = await readContactsFile();
+    const searchedContact = contacts.find(contact => contact.id === contactId);
+
+    return searchedContact || null;
+    // if (searchedContact) {
+    //   console.log('Contact found:'.green);
```
Comment (AdamChodowiecGoIt91) [inline models/contacts.js]: remove comments

File: models/contacts.js [hunk 4]
```diff
@@ -1,19 +1,123 @@
...
+    console.error('Error occured when trying to show contacts:'.red, error);
+  }
+};
+
+const getContactById = async contactId => {
+  try {
+    const contacts = await readContactsFile();
+    const searchedContact = contacts.find(contact => contact.id === contactId);
+
+    return searchedContact || null;
```
Comment (AdamChodowiecGoIt91) [inline models/contacts.js]: you dont need `|| null;` condition. searchedContact will be falsy already

File: models/contacts.js [hunk 5]
```diff
@@ -1,19 +1,123 @@
...
+};
+
+const removeContact = async contactId => {
+  try {
+    const contacts = await readContactsFile();
+    const filteredContacts = contacts.filter(contact => contact.id !== contactId);
+
+    if (filteredContacts.length < contacts.length) {
+      await writeContactsFile(filteredContacts);
+      // console.log(`Contact with ID ${contactId} removed.`.green);
```
Comment (AdamChodowiecGoIt91) [inline models/contacts.js]: Just remove all commented code please :)

File: routes/api/contacts.js [hunk 6]
```diff
@@ -1,25 +1,187 @@
...
+        message: 'Problem with validation',
+        errorDetails: error.details,
+      });
+    }
+    // ---- validation END
+
+    // ---- old ver. START
+    // if (!name || !email || !phone) {
+    //   res.status(400).json({
+    //     status: 'error',
```
Comment (AdamChodowiecGoIt91) [inline routes/api/contacts.js]: comments

**Comment by MarcinNieradka** [comment]
Thank you for feedback. All has been corrected :)

**Pull request merged**